datasets/pg19/pg19.py

```diff
@@ -0,0 +1,179 @@
+"""TODO(pg19): Add a description here."""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+import json
+import os
+from operator import itemgetter
+
+import requests
+
+import nlp
+
+
+# TODO(pg19): BibTeX citation
+_CITATION = """\
+@article{raecompressive2019,
+  author = {Rae, Jack W and Potapenko, Anna and Jayakumar, Siddhant M and
+            Hillier, Chloe and Lillicrap, Timothy P},
+  title = {Compressive Transformers for Long-Range Sequence Modelling},
+  journal = {arXiv preprint},
+  url = {https://arxiv.org/abs/1911.05507},
+  year = {2019},
+}
+
+"""
+
+# TODO(pg19):
+_DESCRIPTION = """\
+This repository contains the PG-19 language modeling benchmark.
+It includes a set of books extracted from the Project Gutenberg books library, that were published before 1919.
+It also contains metadata of book titles and publication dates.
```

Comment:
Also, do you think you can add a word on the preprocessing of the dataset? I think you can add most of the information from: https://github.com/deepmind/pg19 actually.

All these elements on biases and limitations are important for all the users of the dataset.

```
PG-19 is over double the size of the Billion Word benchmark [2] and contains documents that are 20X longer, on average, than the WikiText long-range language modelling benchmark [3].

Books are partitioned into a train, validation, and test set. Book metadata is stored in metadata.csv which contains (book_id, short_book_title, publication_date).

Unlike prior benchmarks, we do not constrain the vocabulary size --- i.e. mapping rare words to an UNK token --- but instead release the data as an open-vocabulary benchmark. The only processing of the text that has been applied is the removal of boilerplate license text, and the mapping of offensive discriminatory words as specified by Ofcom [4] to placeholder tokens. Users are free to model the data at the character-level, subword-level, or via any mechanism that can model an arbitrary string of text.

To compare models we propose to continue measuring the word-level perplexity, by calculating the total likelihood of the dataset (via any chosen subword vocabulary or character-based scheme) divided by the number of tokens --- specified below in the dataset statistics table.

One could use this dataset for benchmarking long-range language models, or use it to pre-train for other natural language processing tasks which require long-range reasoning, such as LAMBADA [5] or NarrativeQA [6]. We would not recommend using this dataset to train a general-purpose language model, e.g. for applications to a production-system dialogue agent, due to the dated linguistic style of old texts and the inherent biases present in historical writing.
```